evals/benchmark/README.md

```diff
@@ -12,12 +12,171 @@ This Tool provides a microservice benchmarking framework that uses YAML configur
+```
+
+#### 3. Deploy with Multiple Instances
+
+To effectively deploy your workload across multiple nodes, follow these steps.
+
+##### Preload Shared Models
+Donwloading models simultaneously across multiple instances can overload resources. To prevent resource exhaustion, it's recommended to preload the models in advance.
+
+```bash
+pip install -U "huggingface_hub[cli]"
+sudo mkdir -p /mnt/models
+sudo chmod 777 /mnt/models
+huggingface-cli download --cache-dir /mnt/models Intel/neural-chat-7b-v3-3
+export MODELDIR=/mnt/models
+```
+Once the models are downloaded, you can consider the following methods for sharing them across nodes:
+- Persistent Volume Claim (PVC): This is recommended approach for production setups. For more details on using PVC, refer to [PVC](https://github.com/opea-project/GenAIInfra/blob/main/helm-charts/README.md#using-persistent-volume).
+- Local Host Path: For simpler testing, ensure that each node involved in the deployment follows the steps above to locally prepare the modes. After preparing the models, use `--set global.modelUseHostPath=${MODELDIR}` in the deployment command.
+
+##### Get the Chart-Defined Service Names
+These are the logical services defined in the Helm chart, such as `retriever-usvc`, `tei`, and `teirerank`, you can retrieve the full list with:
+```bash
+helm show readme opea/chatqna
+```
+##### Get the Node Names
+Get the node name for each of the nodes for deployment.
+```bash
+kubectl get nodes --show-labels
+```
+
+##### Configure Multiple Instances
+
+Now you can configure the number of instances for each service by creating a custom YAML file (e.g., `cluster.yaml`). Add the desired chart-defined service name to your YAML file as shown below:
+```
+# cluster.yaml - Example configuration for multiple instances
+tgi:                          # Chart-defined service name
+  replicaCount: 12            # Number of instances
+  evenly_distributed: true    # Distribute instances evenly across hardware nodes
+  affinity:                   # Schedule pods on specific nodes
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: kubernetes.io/hostname
+                operator: In
+                values:
+                  - node1
+                  - node2
+```
+##### Deploy with Your Custom Configuration
+Use the following command to deploy the workload with multiple instances:
+
+```bash
+helm install chatqna opea/chatqna \
+  --set global.HUGGINGFACEHUB_API_TOKEN=${HFTOKEN} \
+  --set global.modelUseHostPath=${MODELDIR} \
+  -f chatqna/gaudi-values.yaml -f cluster.yaml
+```
+
+#### 4. Deploy with a Specific Namespace
+
+To deploy the workload in a specific namespace:
+
+```bash
+export NAMESPACE=“insert-your-workload-namespace”
+helm install chatqna opea/chatqna \
+  --set global.HUGGINGFACEHUB_API_TOKEN=${HFTOKEN} \
+  -f chatqna/gaudi-values.yaml \
+  --namespace ${NAMESPACE} --create-namespace
```

Comment:
is there a problem if the namespace already exists in? Perhaps say "create if it does not exist"

Reply:
`--create-namespace                           create the release namespace if not present`, Refer to the "helm install --help", there is no need to worry.